### PR TITLE
feat(ARC-2161): add checkbox to blade

### PR DESCRIPTION
**Pull request by RobbeBierebeeck**

### CODE REVIEW

File: src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx [hunk 1]
```diff
@@ -22,9 +27,14 @@ const PersonalInfoBlade: FC<PersonalInfoBladeBladeProps> = ({
 	currentLayer,
 	refetch,
 }) => {
+	const user = useSelector(selectUser);
 	const { tText } = useTranslation();
 	const dispatch = useAppDispatch();
+	const { data: preferences } = useGetNewsletterPreferences(user?.email);
 
+	const [newsLetterChecked, setNewsLetterChecked] = useState<boolean>(
```
Comment (bertyhell) [inline src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx:35]: keep consistant naming with existing functions
```suggestion
	const [newsletterChecked, setNewsletterChecked] = useState<boolean>(
```

File: src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx [hunk 2]
```diff
@@ -22,9 +27,14 @@ const PersonalInfoBlade: FC<PersonalInfoBladeBladeProps> = ({
 	currentLayer,
 	refetch,
 }) => {
+	const user = useSelector(selectUser);
 	const { tText } = useTranslation();
 	const dispatch = useAppDispatch();
+	const { data: preferences } = useGetNewsletterPreferences(user?.email);
```
Comment (bertyhell) [inline src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx:33]:
```suggestion
	const { data: preferences } = useGetNewsletterPreferences(user?.email);
	const shouldRenderNewsletterCheckbox = preferences?.newsletter || false;
```

File: src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx [hunk 3]
```diff
@@ -56,6 +73,17 @@ const PersonalInfoBlade: FC<PersonalInfoBladeBladeProps> = ({
...
+	const renderCheckbox = (renderCheckbox: boolean) => {
+		return !renderCheckbox ? (
+			<Checkbox
+				className={styles['c-personal-info-blade__checkbox']}
+				checked={newsLetterChecked}
+				label={tHtml('schrijf je in voor de nieuwsbrief')}
+				onClick={() => setNewsLetterChecked((prevState) => !prevState)}
+			/>
+		) : null;
+	};
```
Comment (bertyhell) [inline src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx:85]: having a function and a variable be the same name is a bit confusing
also this function has access to the state variables, so you can just make all this logic internal:
```suggestion
	const renderCheckbox = () => {
		if (!shouldRenderNewsletterCheckbox) {
			return null;
		}
		return (
			<Checkbox
				className={styles['c-personal-info-blade__checkbox']}
				checked={newsLetterChecked}
				label={tHtml('schrijf je in voor de nieuwsbrief')}
				onClick={() => setNewsLetterChecked((prevState) => !prevState)}
			/>
		);
	};
```

File: src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx [hunk 4]
```diff
@@ -73,24 +101,30 @@ const PersonalInfoBlade: FC<PersonalInfoBladeBladeProps> = ({
...
-						<Button
-							label={tText(
-								'modules/navigation/components/personal-info-blade/personal-info-blade___verstuur'
-							)}
-							variants={['block', 'text', 'dark']}
-							onClick={onSendRequests}
-							className={styles['c-personal-info-blade__send-button']}
-						/>
+						<>
+							{renderCheckbox(preferences?.newsletter || false)}
```
Comment (bertyhell) [inline src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx:118]:
```suggestion
							{renderCheckbox()}
```

**Review by bertyhell — APPROVED** [review]

File: src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx (hunk 3 above)
Comment (ian-emsens-shd) [reply src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx]: + avoid render functions in general 😅
Their presence usually indicates you need a "domain component" to map your data structure to dumber ui components.
Screenshot 2024-06-18 at 11 49 04

**Review by ian-emsens-shd — APPROVED** [review]

**Pull request merged**